src/app/shared/services/transaction.service.ts

```diff
@@ -93,35 +112,76 @@ export class TransactionService implements OnDestroy {
     }
 
     this.wallet = wallet;
-    this.currentTransactions = this.readFromStore();
+
+    const storedData = this.readFromStore();
+    this.currentTransactions = storedData.currentTransactions;
+    this.droppedTransactions = storedData.droppedTransactions;
+
     this.currentTransactionsSubject.next(this.currentTransactions);
+    this.droppedTransactionsSubject.next(this.droppedTransactions);
+    this.currentTransactions.forEach(transaction => this.trackOrphanedTransaction(transaction));
   }
 
   removeTransaction(transaction: Transaction): void {
     this.currentTransactions = this.currentTransactions
       .filter(_transaction => transaction.transactionHash !== _transaction.transactionHash);
-    this.saveToStore(this.currentTransactions);
+    this.trackedTransactionHashes.splice(this.trackedTransactionHashes.indexOf(transaction.transactionHash), 1);
+    this.updateStore();
     this.currentTransactionsSubject.next(this.currentTransactions);
   }
 
+  removeDroppedTransactions(identifier: string) {
+    this.droppedTransactions = this.droppedTransactions.filter(transaction => transaction.identifier !== identifier);
+    this.updateStore();
+    this.droppedTransactionsSubject.next(this.droppedTransactions);
+  }
+
+  async trackOrphanedTransaction(transaction: Transaction): Promise<void> {
+    if (this.trackedTransactionHashes.includes(transaction.transactionHash)) {
+      return;
+    }
+
+    this.trackedTransactionHashes.push(transaction.transactionHash);
+
+    try {
+      await this.getTransactionReceipt(transaction.transactionHash);
+    } catch (error) {
+      if (error === DroppedTransactionError) {
```

Comment:
what else? is it expected behaviour to keep application running in such case?

Reply:
If error !== DroppedTransactionError then we just stop tracking this transaction. This will unlock button related with transaction and user will be able to perform action again.